machine/machine.go

```diff
@@ -42,7 +42,7 @@ var (
 	coreRegExp = regexp.MustCompile(`(?m)^core id\s*:\s*([0-9]+)$`)
 	nodeRegExp = regexp.MustCompile(`(?m)^physical id\s*:\s*([0-9]+)$`)
 	// Power systems have a different format so cater for both
-	cpuClockSpeedMHz     = regexp.MustCompile(`(?:cpu MHz|clock)\s*:\s*([0-9]+\.[0-9]+)(?:MHz)?`)
+	cpuClockSpeedMHz     = regexp.MustCompile(`(?i)(?:cpu MHz|clock)\s*:\s*([0-9]+\.[0-9]+)(?:MHz)?`)
```

Comment:
I can see an issue with SI prefixes here: `mHz` that would match to case-insensitive `MHz` regexp represents milihertz, that is one thousandth of hertz. `mhz` that would match newly proposed regexp is incorrect (because of lack of capital H). 
Could you rather allow both `cpu` and `CPU` rather than make regexp case-insensitive, please? As far as I understand it should fix your issue.